nightfall-optimist/src/event-handlers/new-current-proposer.mjs

```diff
@@ -18,10 +18,11 @@ async function newCurrentProposerEventHandler(data, args) {
   try {
     logger.info(`Proposer Handler - Current proposer is ${currentProposer}`);
     // remember the current proposer.  We don't store it in the DB as it's an
-    // ephemeral thing. Instead, we have a nice object to remember it. The
+    // ephemeral thing. Instead, we have an object to remember it. The
     // object is instantiated at the 'main()' level, so will stay in scope even
     // when this handler exits.
     proposer.address = currentProposer;
+    proposer.proposers.unshift(currentProposer); // useful if we need to revert the proposer because of a chain reorg.
```

Comment:
This global `proposers` array will grow unbounded with each proposer rotation and contain a large number of duplicates. Given the relative rarity of a chain re-org I would expect this proposers array to become very large which is both a potential memory constraint and inefficient (AFAIK unshift is O(n) ).

Noting that to `removeNewCurrentProposer` we need a way to access past currentProposers; alternate ways could be to: 
- Use the on-chain `proposers[currentProposer].previousAddress` to step back when a `NewCurrentProposer` is removed or,
- Change the `NewCurrentProposer` event to emit the `LinkedAddress` struct which contains the `previousAddress`, `currentAddress` and `nextAddress`.

Reply:
Yes, this isn't optimal.

Reply:
Now I read the currentProposer LinkedAddress variable.  That should be the actual value I want because it will have reverted to whatever currentProposer was before the chain reorg.